modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/clip/ShareClipFragment.kt

```diff
@@ -81,22 +89,38 @@ class ShareClipFragment : BaseDialogFragment() {
         savedInstanceState: Bundle?,
     ) = ComposeView(requireActivity()).apply {
         val platforms = SocialPlatform.getAvailablePlatforms(requireContext())
-        val assetController = BackgroundAssetController.create(requireContext(), shareColors)
-        val listener = shareListenerFactory.create(this@ShareClipFragment, viewModel, assetController, args.source)
+        val assetController = BackgroundAssetController.create(requireContext().applicationContext, shareColors)
+        val listener = ShareClipListener(this@ShareClipFragment, viewModel, assetController, args.source)
+
         setContent {
             val uiState by viewModel.uiState.collectAsState()
+            val snackbarHostState = remember { SnackbarHostState() }
+
             ShareClipPage(
                 episode = uiState.episode,
                 podcast = uiState.podcast,
                 clipRange = uiState.clipRange,
                 playbackProgress = uiState.playbackProgress,
                 isPlaying = uiState.isPlaying,
+                isSharing = uiState.isSharing,
                 useEpisodeArtwork = uiState.useEpisodeArtwork,
                 platforms = platforms,
                 shareColors = shareColors,
                 assetController = assetController,
                 listener = listener,
+                snackbarHostState = snackbarHostState,
             )
+
+            LaunchedEffect(Unit) {
+                viewModel.snackbarMessages.collect { message ->
+                    val text = when (message) {
+                        is SnackbarMessage.SharingResponse -> message.message
+                        is SnackbarMessage.PlayerIssue -> getString(LR.string.podcast_episode_playback_error)
```

Comment:
This error looks good, but I have a minor comment on the UI behaviour. I'm not sure if it's possible, but it would be nice if the play button changed to a pause button as soon as it's tapped. At the moment, there is a delay, and it can feel like the button hasn't been pressed.

Reply:
I agree with you. I just didn't want to risk our managed state get out of sync with the ExoPlayer. But it probably is possible to use something like 'isLoading' for one of ExoPlayer's callbacks. I'll look into it in a separate PR because it might put me on a journey of its own.